lib/src/auth/user_model.dart

```diff
@@ -1,28 +1,51 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'user_model.g.dart';
+
+@JsonSerializable()
 class User {
   String email;
   String? firstName;
   String? lastName;
-  String password;
+  String? name;
+  String? password;
+  String? createdAt;
+  String? updatedAt;
+  String? id;
+  List<dynamic>? roles;
 
   User({
     required this.email,
-    required this.password,
+    this.password,
+    this.name,
     this.firstName,
     this.lastName,
+    this.createdAt,
+    this.updatedAt,
+    this.id,
+    this.roles,
   });
 
-  Map<String, dynamic> get loginParams {
-    return {'email': email, 'password': password};
-  }
+  Map<String, dynamic> get loginParams => {
+        'email': email,
+        'password': password,
+      };
 
-  Map<String, dynamic> get registerParams {
-    return {
-      'email': email,
-      'firstName': firstName,
-      'lastName': lastName,
-      'password': password,
-    };
-  }
+  Map<String, dynamic> get registerParams => {
+        'email': email,
+        'firstName': firstName,
+        'lastName': lastName,
+        'password': password,
+      };
 
   Map<String, dynamic> get resetPasswordParams => {'email': email};
+
+  Map<String, dynamic> get updateParams => {
+        'firstName': firstName ?? '',
+        'lastName': lastName ?? '',
+      };
```

Comment:
Correct me if I'm wrong but the methods `loginParams` `registerParams` `resetPasswordParams` `updateParams` were created to extract data to make API requests, right? If so then I would advice them to be RequestModels.

```
@JsonSerializable()
class RegisterUserRequest {
  String email;
  String firstName;
  String lastName;
  String password;

  RegisterUserRequest({
    required this.email,
    required this.firstName,
    required this.lastName,
    required this.password,   
  });

   factory RegisterUserRequest.fromJson(Map<String, dynamic> json) => _$RegisterUserRequestFromJson(json);

  Map<String, dynamic> toJson() => _$RegisterUserRequestToJson(this);
  ```

Reply:
Good idea. Done!